### PR TITLE
Improve clarity in troubleshooting doc

### PR DESCRIPTION
**Pull request by PoorvajaRayas**
Improved the documentation to make it clearer

### CODE REVIEW

File: docs/source/troubleshooting.md [hunk 1]
```diff
@@ -1,22 +1,22 @@
 # Troubleshooting
 
-When troubleshooting, you may see unexpected behaviors or receive an error
+While troubleshooting, you may see unexpected behaviors or receive an error
```
Comment (minrk) [inline docs/source/troubleshooting.md:3]: I would keep 'when'

File: docs/source/troubleshooting.md [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
 message. This section provides links for identifying the cause of the
 problem and how to resolve it.
 
 ## Behavior
 
 ### JupyterHub proxy fails to start
 
-If you have tried to start the JupyterHub proxy and it fails to start:
+If you have tried to start the JupyterHub proxy and it fails to run:
```
Comment (minrk) [inline docs/source/troubleshooting.md:11]: I think 'start' is better here

File: docs/source/troubleshooting.md [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
 
 - check if the JupyterHub IP configuration setting is
   `c.JupyterHub.ip = '*'`; if it is, try `c.JupyterHub.ip = ''`
 - Try starting with `jupyterhub --ip=0.0.0.0`
 
 **Note**: If this occurs on Ubuntu/Debian, check that you are using a
-recent version of [Node](https://nodejs.org). Some versions of Ubuntu/Debian come with a version
-of Node that is very old, and it is necessary to update Node.
+recent version of [Node](https://nodejs.org). Some versions of Ubuntu/Debian come with an older version
```
Comment (minrk) [inline docs/source/troubleshooting.md:18]: This dramatically weakens the language - we want to communicate that it is very old. At the very least, it should say that it's "old" not "older"

File: docs/source/troubleshooting.md [hunk 4]
```diff
@@ -37,11 +37,11 @@ to the config file, `jupyterhub_config.py`.
 When nothing is given for these lists, there will be no admins, and all users
 who can authenticate on the system (i.e. all the Unix users on the server with
 a password) will be allowed to start a server. The allowed username set lets you limit
-this to a particular set of users, and admin_users lets you specify who
+this to a particular set of users, and admin_users let you specify who
```
Comment (minrk) [inline docs/source/troubleshooting.md:40]: `admin_users` is a singular configuration value, so it should be `lets` not `let.

File: docs/source/troubleshooting.md [hunk 5]
```diff
@@ -37,11 +37,11 @@ to the config file, `jupyterhub_config.py`.
...
 among them may use the admin interface (not necessary, unless you need to do
 things like inspect other users' servers or modify the user list at runtime).
 
-### JupyterHub Docker container not accessible at localhost
+### What to do when JupyterHub Docker container is not accessible at localhost?
```
Comment (minrk) [inline docs/source/troubleshooting.md:44]: This doesn't fit most of the other headings, which are statements about what's going wrong.

File: docs/source/troubleshooting.md [hunk 6]
```diff
@@ -87,7 +87,7 @@ sudo MY_ENV=abc123 \
 
 ### Error 500 after spawning my single-user server
 
-You receive a 500 error when accessing the URL `/user/<your_name>/...`.
```
Comment (minrk) [inline docs/source/troubleshooting.md:90]: I would call it a '500 error' not 'error 500'

File: docs/source/troubleshooting.md [hunk 7]
```diff
@@ -71,7 +71,7 @@ After successfully logging in to JupyterHub with a compatible authenticator, I g
 This issue occurs when the authenticator requires a local system user to exist. In these cases, you need to use a spawner
 that does not require an existing system user account, such as `DockerSpawner` or `KubeSpawner`.
 
-### How can I run JupyterHub with sudo but use my current env vars and virtualenv location?
+### How can I run JupyterHub with sudo but use my current environment variables and virtualenv location?
```
Comment (minrk) [inline docs/source/troubleshooting.md:74]: 👍

File: docs/source/troubleshooting.md [hunk 8]
```diff
@@ -167,7 +167,7 @@ When your whole JupyterHub sits behind an organization proxy (_not_ a reverse pr
 
 [JupyterHub services](https://jupyterhub.readthedocs.io/en/stable/reference/services.html) allow processes to interact with JupyterHub's REST API. Example use-cases include:
 
-- **Secure Testing**: provide a canonical Jupyter Notebook for testing production data to reduce the number of entry points into production systems.
+- **Secure Testing**: provides a canonical Jupyter Notebook for testing production data to reduce the number of entry points into production systems.
```
Comment (minrk) [inline docs/source/troubleshooting.md:170]: `provide` is correct here

File: docs/source/troubleshooting.md [hunk 9]
```diff
@@ -324,7 +324,7 @@ Or use syslog:
 
 ### Toree integration with HDFS rack awareness script
 
-The Apache Toree kernel will have an issue when running with JupyterHub if the standard HDFS
+The Apache Toree kernel will have an issue while running with JupyterHub if the standard HDFS
```
Comment (minrk) [inline docs/source/troubleshooting.md:327]: I think 'when' is better here

File: docs/source/troubleshooting.md [hunk 10]
```diff
@@ -371,7 +371,7 @@ You can also tail logs to view them in real-time using the `-f` option:
 ## Troubleshooting commands
 
 The following commands provide additional detail about installed packages,
-versions, and system information that may be helpful when troubleshooting
+versions, and system information that may be helpful while troubleshooting
```
Comment (minrk) [inline docs/source/troubleshooting.md:374]: Again, I think when is better here

**Review by minrk — COMMENTED** [review]
Thanks for your contribution! I think several of these changes should be reverted, but some are helpful. I've left suggestions inline.

File: docs/source/troubleshooting.md (hunk 5 above)
Comment (PoorvajaRayas) [reply docs/source/troubleshooting.md:44]: @minrk I think it was started off with answering questions which then moved on to stating the problems in the form of headings and providing solutions . So either I have to change those initial questions into headings to make everything look well structured or the other way around.

Comment (minrk) [reply docs/source/troubleshooting.md:44]: Good point. I think for the most part, the headings under 'behavior' should be _a behavior_ where possible, (i.e. "JupyterHub Docker container is not accessible at localhost"). If you see ways to rephrase one or more of the questions to match, that would be appropriate.

But not all situations quite fit that.

**Comment by minrk** [comment]
Great job, thank you!

**Comment by PoorvajaRayas** [comment]
@minrk Thanks a lot for merging! :)

**Pull request merged**